packages/react/src/components/TableHandles/ExtendButton/ExtendButton.tsx

```diff
@@ -0,0 +1,181 @@
+import {
+  DefaultInlineContentSchema,
+  DefaultStyleSchema,
+  InlineContentSchema,
+  PartialTableContent,
+  StyleSchema,
+} from "@blocknote/core";
+import { MouseEvent as ReactMouseEvent, useEffect, useState } from "react";
+
+import { useComponentsContext } from "../../../editor/ComponentsContext.js";
+import { TableHandleProps } from "../TableHandleProps.js";
+
+const getContentWithAddedRows = <
+  I extends InlineContentSchema,
+  S extends StyleSchema
+>(
+  content: PartialTableContent<I, S>,
+  rowsToAdd = 1
+): PartialTableContent<I, S> => {
+  const newRow: PartialTableContent<I, S>["rows"][number] = {
+    cells: content.rows[0].cells.map(() => []),
+  };
+  const newRows: PartialTableContent<I, S>["rows"] = [];
+  for (let i = 0; i < rowsToAdd; i++) {
+    newRows.push(newRow);
+  }
+
+  return {
+    type: "tableContent",
+    columnWidths: content.columnWidths
+      ? [...content.columnWidths, ...newRows.map(() => undefined)]
+      : undefined,
+    rows: [...content.rows, ...newRows],
+  };
+};
+
+const getContentWithAddedCols = <
+  I extends InlineContentSchema,
+  S extends StyleSchema
+>(
+  content: PartialTableContent<I, S>,
+  colsToAdd = 1
+): PartialTableContent<I, S> => {
+  const newCell: PartialTableContent<I, S>["rows"][number]["cells"][number] =
+    [];
+  const newCells: PartialTableContent<I, S>["rows"][number]["cells"] = [];
+  for (let i = 0; i < colsToAdd; i++) {
+    newCells.push(newCell);
+  }
+
+  return {
+    type: "tableContent",
+    columnWidths: content.columnWidths || undefined,
+    rows: content.rows.map((row) => ({
+      cells: [...row.cells, ...newCells],
+    })),
+  };
+};
+
+export const ExtendButton = <
+  I extends InlineContentSchema = DefaultInlineContentSchema,
+  S extends StyleSchema = DefaultStyleSchema
+>(
+  props: Pick<
+    TableHandleProps<I, S>,
+    "block" | "editor" | "orientation" | "freezeHandles" | "unfreezeHandles"
+  >
+) => {
+  const Components = useComponentsContext()!;
+
+  const [editingState, setEditingState] = useState<{
+    startPos: number;
+    numOriginalCells: number;
+    clickOnly: boolean;
+  } | null>(null);
+
+  const mouseDownHandler = (event: ReactMouseEvent) => {
+    props.freezeHandles();
+    setEditingState({
+      startPos: props.orientation === "row" ? event.clientX : event.clientY,
```

Comment:
Yeah it is kind of confusing, you could justify either the bottom or right placement as being the `row` orientation. Imo it makes most sense if the button on the right is for `row`, since it extends the rows in the table and the table handle for the `row` orientation is on the left (so it's consistent with `row` being left/right and `col` being top/bottom).